demo/src/main/java/com/braintreepayments/api/demo/DemoActivity.java

```diff
@@ -3,22 +3,71 @@
 import android.os.Bundle;
 
 import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 
+import com.braintreepayments.api.BrowserSwitchClient;
+import com.braintreepayments.api.BrowserSwitchException;
+import com.braintreepayments.api.BrowserSwitchOptions;
+import com.braintreepayments.api.BrowserSwitchResult;
+
 public class DemoActivity extends AppCompatActivity {
 
     private static final String FRAGMENT_TAG = DemoFragment.class.getSimpleName();
 
+    @VisibleForTesting
+    BrowserSwitchClient browserSwitchClient = null;
+
+    private String returnUrlScheme;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        Bundle args = getIntent().getExtras();
+        if (args != null) {
+            returnUrlScheme = args.getString("returnUrlScheme");
+        }
+
+        browserSwitchClient = new BrowserSwitchClient();
+
         FragmentManager fm = getSupportFragmentManager();
         if (fm.findFragmentByTag(FRAGMENT_TAG) == null) {
```

Comment:
Would it make sense to use the `findDemoFragment` method here?
```suggestion
        if (findDemoFragment() == null) {
```

Reply:
Also, this is a nit-pick, but it might make sense to call it `getDemoFragment` instead of `findDemoFragment`. You can take or leave that suggestion 🙂